src/on-chain-value.ts

```diff
@@ -27,45 +60,73 @@ export const chainValueDataPoint: APIGatewayProxyHandler = async (event): Promis
     };
   }
 
-  const provider = config.providers[chainId];
-  if (!provider) {
+  const db = await initDb();
+  if (db === undefined) {
     return {
       statusCode: 500,
       headers: config.headers,
-      body: makeError("We don't have a provider for that chainId"),
+      body: makeError('An error has occurred while trying to initialize the database'),
     };
   }
 
-  const dapiServerAddress = contracts.DapiServer[chainId];
-  if (!dapiServerAddress) {
+  const currentDataFeedId = dapiName
+    ? await getDataFeedIdFromDapiName(dapiName, db)
+    : dataFeedId ??
+      ethers.utils.keccak256(ethers.utils.solidityPack(['address', 'bytes32'], [airnodeAddress, templateId]));
+
+  if (!currentDataFeedId) {
     return {
       statusCode: 500,
       headers: config.headers,
-      body: makeError("We don't have a dapiServer for that chainId"),
+      body: makeError('DataFeedId not found'),
     };
   }
 
-  const voidSigner = new ethers.VoidSigner(
-    ethers.constants.AddressZero,
-    new ethers.providers.JsonRpcProvider(provider)
-  );
-
-  const dapiServer = DapiServer__factory.connect(dapiServerAddress, voidSigner);
-
-  const readDataFeedId =
-    dataFeedId ??
-    ethers.utils.keccak256(ethers.utils.solidityPack(['address', 'bytes32'], [airnodeAddress, templateId]));
-
   const operation = async () =>
-    await (dapiName ? dapiServer.readDataFeedWithDapiName(dapiName!) : dapiServer.readDataFeedWithId(readDataFeedId));
-  const [err, beaconResponse] = await go(operation, { timeoutMs: 5_000, retries: 2 });
+    db.query(
+      `
+      SELECT jsonb_path_query_array(event_data -> 'parsedLog' -> 'args' , '$[1 to 2]') as "data"
```

Comment:
> `jsonb_path_query_array(`

Very nice... I wanted to suggest this but thought it might make things more complex, so I kept quiet. Well done :)